src/serialize.rs

```diff
@@ -95,22 +92,7 @@ impl<'de> Deserialize<'de> for Prefix {
                 where
                     E: de::Error,
                 {
-                    let mut split = s.split('/');
-
-                    let hex_str = split
-                        .next()
-                        .ok_or_else(|| de::Error::custom("`str::split` logic error"))?;
-                    let bit_count = split
-                        .next()
-                        .ok_or_else(|| de::Error::custom("no `/` symbol encountered"))?;
-
-                    let k: [u8; 32] = hex::FromHex::from_hex(hex_str)
-                        .map_err(|_| de::Error::custom("invalid 32 byte hex string"))?;
-                    let bit_count = bit_count
-                        .parse::<usize>()
-                        .map_err(|_e| de::Error::custom("bit_count is not a valid `usize`"))?;
-
-                    Ok(Prefix::new(bit_count, XorName(k)))
+                    Ok(Prefix::from_str(s).unwrap())
```

Comment:
This looks like a panic. Can we look at returning even interior errors here instead

Reply:
Thanks for spotting this David! Will turn it into proper error...